### ISSUE TITLE
Article "Extending WC-Admin Reports" needs an update

### ISSUE THREAD

**Issue by daigo75** [woocommerce/woocommerce-admin#4491]
The example described on the article [Extending WC-Admin Reports](https://woocommerce.wordpress.com/2020/02/20/extending-wc-admin-reports/) doesn't seem to work for the dashboard. It's possible to show the currency filter widget on the report (see https://prnt.sc/ssuwiy), as indicated in your "SQL Modification" example, but that filter doesn't do anything. Specifically:
1. Changing the currency doesn't trigger a refresh, like it happens on the other reports.
2. The dashboard fires a dozen Ajax requests to fetch its sub-sections. None of the calls include the currency from the filter.

In short, it looks like some more work has to be done on the dashboard to add the currency filter.

Expected behaviour
--
1. Changing the currency must trigger a refresh of the data, like it happens on the other reports.
2. The selected currency must be passed with all the calls performed by the dashboard report.

Priority
--
As soon as possible, to complete the integration process originally described on the development blog.

**Comment by rivkadr** [comment]
Any update on this? I'm currently trying to use the SQL Modification Example, and it's not working at all (throws this error: SyntaxError: import declarations may only appear at top level of a module)

**Comment by psealock** [comment]
Thanks for the comment @daigo75, the Dashboard was lacking an ability to react to changes in the url query.

https://github.com/woocommerce/woocommerce-admin/pull/4758 Makes sure query parameters are passed to elements of the dashboard so that new requests can be made that reflect the added parameters of the filters. Next, I will look into how those parameters are received and make sure they are handled properly.

@rivkadr I haven't seen that error myself. Is it possible you are declaring your imports in the wrong spot? Are you running `npm start` in the extension repo?

**Comment by daigo75** [comment]
@psealock Thanks for the update. Indeed, I had noticed that the dashboard is made of "sub-reports", I guessed that the "parent" had to pass the query arguments to them. 👍

**Comment by psealock** [comment]
>Next, I will look into how those parameters are received and make sure they are handled properly.

I went through the Overview and confirmed that endpoints respect the added `currency` query parameter. As long as the extension filters all the report query args and `woocommerce_analytics_clauses_*` requested by the dashboard, then data will respect the `currency` query argument.

As a result, I think its safe to close this one. Thanks again @daigo75 for bringing it to my attention.

**Comment by thuhoang** [comment]
Hi guy!
Can you give me a complete example code ?
I follow your guide but it does not work

Thanks.
